components/script/dom/subtlecrypto.rs

```diff
@@ -865,15 +882,31 @@ impl SubtleCrypto {
             256 => Handle::Aes256(data.to_vec()),
             _ => return Err(Error::Data),
         };
-        let name = DOMString::from(alg);
-        Ok(CryptoKey::new(
-            &self.global(),
-            KeyType::Secret,
-            extractable,
-            KeyAlgorithm { name },
-            usages,
-            handle,
-        ))
+
+        let name = DOMString::from(alg_name);
```

Comment:
Can we move this into a helper function to reduce duplication?

Reply:
I tried doing that in https://github.com/servo/servo/pull/34092/commits/42bba2d4a1fa86c1adb53b10b4f494625bb7ecb8 - there's still some duplication because I'm not sure how to return a rooted object from a function safely, so we end up with an out parameter to `AesKeyAlgorithm::from_name_and_size` and callers still need to construct the algorithm object themselves.